{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"mount_file_id":"1Q_jTbU6JMj1WFg8rkI5CeF4kwwFLnkUt","authorship_tag":"ABX9TyN2qUrJwGIM3/GP5A9ria78"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"markdown","source":["CHAPTER 1"],"metadata":{"id":"NYD8SNNvjJ_3"}},{"cell_type":"markdown","source":["Each line of code below contains an error. Fix the error and explain in a comment the reason for the error.\n"],"metadata":{"id":"QEyu3YnCntWZ"}},{"cell_type":"markdown","source":["EXERCISES"],"metadata":{"id":"rUKnsgV-j4eh"}},{"cell_type":"markdown","source":["Exercise 1"],"metadata":{"id":"8VfoXx4wkEjQ"}},{"cell_type":"code","source":["Print('Hello world!')"],"metadata":{"id":"L2l_ZcOPnzqg"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["print('Hello world!') # In Python, the function for printing is called \"print\" with a\n","#Python is sensitive to the capitalization of function names, so using \"Print\" with an uppercase \"P\" causes an error because Python doesn't recognize it as the correct function."],"metadata":{"id":"FQJ03sQGkGBP","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1709884465553,"user_tz":-480,"elapsed":9,"user":{"displayName":"Ella Florenz DE JUAN","userId":"00598978747470842528"}},"outputId":"88ca6b49-eedb-468f-d664-eed6932b9e32"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Hello world!\n"]}]},{"cell_type":"code","source":["print'Hello world!')"],"metadata":{"id":"eknSzOP6kKUw"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["print('Hello world!')#The error occurs because the print statement is missing opening parentheses.\n","#print is a function and requires parentheses around its arguments."],"metadata":{"id":"Ohyp-e7QoJgD","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1709884469441,"user_tz":-480,"elapsed":3,"user":{"displayName":"Ella Florenz DE JUAN","userId":"00598978747470842528"}},"outputId":"253b7275-a24d-44ad-a0d9-27b66889d6b2"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Hello world!\n"]}]},{"cell_type":"code","source":["print('Hello world!'')"],"metadata":{"id":"OsLH7xspo4qK"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["print('Hello world!') #The error was caused by using both single and double quotes to enclose the string 'Hello world!', which is not allowed in Python, so removing the extra single quotes to fixed the issue."],"metadata":{"id":"mU-9AFAmo6IQ","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1709884476556,"user_tz":-480,"elapsed":324,"user":{"displayName":"Ella Florenz DE JUAN","userId":"00598978747470842528"}},"outputId":"767485a5-7d15-44fb-c5ed-9ba771eeaaf1"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Hello world!\n"]}]},{"cell_type":"markdown","source":["Exercise 2\n","\n","What is the output of the code below? [PEW]"],"metadata":{"id":"_8N_0yHqpObw"}},{"cell_type":"code","source":["x = 43\n","x = x + 1\n","print(x) #44 is the answer"],"metadata":{"id":"WH4D31_3pbta","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1709884485498,"user_tz":-480,"elapsed":349,"user":{"displayName":"Ella Florenz DE JUAN","userId":"00598978747470842528"}},"outputId":"7e3c2a54-36fb-4406-aa7c-02596dc5e34f"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["44\n"]}]},{"cell_type":"markdown","source":["Exercise 3\n","\n","The code below contains an error. Fix the error and explain in a comment the reason for the error."],"metadata":{"id":"dtwcdzTspiWm"}},{"cell_type":"code","source":["x = 43\n","x = x + 1\n","print(X)"],"metadata":{"id":"1hwy0ynSpspI"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["x = 43\n","x = x + 1\n","print(x)  # The variable 'x' is defined with a lowercase 'x', so it should be printed with a lowercase 'x'. Python is case-sensitive, so 'X' and 'x' are treated as different variables.\n","#so i change the variable 'X' to 'x' to fix the error"],"metadata":{"id":"sR30a65Iptp7","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1709884488334,"user_tz":-480,"elapsed":333,"user":{"displayName":"Ella Florenz DE JUAN","userId":"00598978747470842528"}},"outputId":"b6129d75-e94b-48a5-9a7a-3197b3e72d7c"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["44\n"]}]}]}